### PR TITLE
Clear nofity socket from containerd env

### LINKED ISSUE #21507: DockerSuite.TestDiffEnsureDefaultDevs failing consistently on systemd system
Related to https://github.com/docker/docker/pull/21465, this test is consistently failing on systemd system it seems.

Quoting @runcom (https://github.com/docker/docker/pull/21465#issuecomment-200774615)

> ping @crosbymichael @LK4D4 @calavera I don't have any clue about who's mounting /run/systemd/notify  (I'm on fedora 23 and docker built from master)
> 
> I get this in every container:
> ```
> # mount | grep /run
> tmpfs on /run/systemd/notify type tmpfs (rw,seclabel,nosuid,nodev,mode=755)
> ```
> it's mounted from the host (same inode also)

```
$ docker run -itd busybox sh
$ docker diff a95
A /run
A /run/systemd
A /run/systemd/notify
```

* Docker version

```
Client:
 Version:      1.11.0-dev
 API version:  1.24
 Go version:   go1.6
 Git commit:   b64627b
 Built:        Thu Mar 24 08:31:20 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.11.0-dev
 API version:  1.24
 Go version:   go1.6
 Git commit:   b64627b
 Built:        Thu Mar 24 08:31:20 2016
 OS/Arch:      linux/amd64
```

* Docker info

```
Containers: 16
 Running: 1
 Paused: 0
 Stopped: 15
Images: 262
Server Version: 1.11.0-dev
Storage Driver: overlay
 Backing Filesystem: extfs
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: bridge null host
Kernel Version: 4.4.5-1-ARCH
Operating System: Arch Linux
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 15.56 GiB
Name: hokkaido
ID: YEDF:TDNS:RY6Q:D6RB:OH7S:EPBP:DBZ5:BAVO:HPEH:ALXT:PLP2:FLDM
Docker Root Dir: /var/lib/docker
Debug mode (client): false
Debug mode (server): true
 File Descriptors: 22
 Goroutines: 47
 System Time: 2016-03-24T11:15:20.211478419+01:00
 EventsListeners: 0
Username: vdemeester
Registry: https://index.docker.io/v1/
Labels:
 type=laptop
```

* Kernel

```
Linux hokkaido 4.4.5-1-ARCH #1 SMP PREEMPT Thu Mar 10 07:38:19 CET 2016 x86_64 GNU/Linux
```

### LINKED ISSUE #21684: Test TestRunEnvironment* fails on Fedora (& systemd?)
same env (and probably relaetd to) as https://github.com/docker/docker/issues/21656, there should be something new|wrong with new runc,containerd,docker

The tests in $SUBJECT now fail on my host because the `env` has one more env variable (which is from systemd socket):
```
----------------------------------------------------------------------
FAIL: docker_cli_run_test.go:785: DockerSuite.TestRunEnvironment

docker_cli_run_test.go:815:
    c.Fatalf("Wrong environment: should be %d variables, not: %q\n", len(goodEnv), strings.Join(actualEnv, ", "))
... Error: Wrong environment: should be 8 variables, not: ", FALSE=true, HOME=/root, HOSTNAME=testing, NOTIFY_SOCKET=/run/systemd/notify, PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin, TRICKY=tri, TRUE=false, cky"



----------------------------------------------------------------------
FAIL: docker_cli_run_test.go:824: DockerSuite.TestRunEnvironmentErase

docker_cli_run_test.go:850:
    c.Fatalf("Wrong environment: should be %d variables, not: %q\n", len(goodEnv), strings.Join(actualEnv, ", "))
... Error: Wrong environment: should be 2 variables, not: "HOME=/root, NOTIFY_SOCKET=/run/systemd/notify, PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"



----------------------------------------------------------------------
FAIL: docker_cli_run_test.go:859: DockerSuite.TestRunEnvironmentOverride

docker_cli_run_test.go:885:
    c.Fatalf("Wrong environment: should be %d variables, not: %q\n", len(goodEnv), strings.Join(actualEnv, ", "))
... Error: Wrong environment: should be 3 variables, not: "HOME=/root2, HOSTNAME=bar, NOTIFY_SOCKET=/run/systemd/notify, PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"



----------------------------------------------------------------------
```

ping @LK4D4  @vdemeester

### PR DESCRIPTION
**Pull request by crosbymichael**
Fixes #21507 
Fixes #21684

Signed-off-by: Michael Crosby crosbymichael@gmail.com

### CODE REVIEW

File: libcontainerd/remote_linux.go [hunk 1]
```diff
@@ -351,11 +352,19 @@ func (r *remote) runContainerdDaemon() error {
...
+
 	cmd := exec.Command(containerdBinary, args...)
 	// redirect containerd logs to docker logs
 	cmd.Stdout = os.Stdout
 	cmd.Stderr = os.Stderr
 	cmd.SysProcAttr = &syscall.SysProcAttr{Setsid: true}
+	cmd.Env = nil
+	// clear the NOTIFY_SOCKET from the env when starting containerd
+	for _, e := range os.Environ() {
+		if !strings.Contains(e, "NOTIFY_SOCKET") {
```
Comment (cpuguy83) [inline libcontainerd/remote_linux.go]: Can we just unset the env in the engine's environment?

Comment (crosbymichael) [inline libcontainerd/remote_linux.go]: Its still read from docker so I don't think we want that and its read after contianerd is setup.

Comment (LK4D4) [inline libcontainerd/remote_linux.go]: I think it's safer to use `strings.HasPrefix` here.

Comment (mlaventure) [inline libcontainerd/remote_linux.go]: @crosbymichael is correct, the env value is used **after** containerd is started in `notifySystem()`

**Comment by mlaventure** [comment]
Looks good.

**Comment by tonistiigi** [comment]
LGTM

**Comment by crosbymichael** [comment]
@LK4D4 fixed

**Comment by LK4D4** [comment]
LGTM

**Pull request merged**